lang/py/src/avro/schema.py

```diff
@@ -887,20 +895,30 @@ def make_bytes_decimal_schema(other_props):
 def make_logical_schema(logical_type, type_, other_props):
   """Map the logical types to the appropriate literal type and schema class."""
   logical_types = {
-    constants.DATE: ('int', DateSchema),
-    # Fixed decimal schema is handled before we get here.
-    constants.DECIMAL: ('bytes', make_bytes_decimal_schema),
-    constants.TIMESTAMP_MICROS: ('long', TimestampMicrosSchema),
-    constants.TIMESTAMP_MILLIS: ('long', TimestampMillisSchema),
-    constants.TIME_MICROS: ('long', TimeMicrosSchema),
-    constants.TIME_MILLIS: ('int', TimeMillisSchema),
+    (constants.DATE, 'int'): DateSchema,
+    (constants.DECIMAL, 'bytes'): make_bytes_decimal_schema,
+    # The fixed decimal schema is handled later by returning None now.
+    (constants.DECIMAL, 'fixed'): lambda x: None,
+    (constants.TIMESTAMP_MICROS, 'long'): TimestampMicrosSchema,
+    (constants.TIMESTAMP_MILLIS, 'long'): TimestampMillisSchema,
+    (constants.TIME_MICROS, 'long'): TimeMicrosSchema,
+    (constants.TIME_MILLIS, 'int'): TimeMillisSchema,
   }
-  literal_type, schema_type = logical_types.get(logical_type, (None, None))
   try:
-    if literal_type == type_:
+    schema_type = logical_types.get((logical_type, type_), None)
+    if schema_type is not None:
       return schema_type(other_props)
-  except SchemaParseException:
-    pass
+
+    expected_types = [literal_type for lt, literal_type in logical_types if lt == logical_type]
+    if expected_types:
+      expected_types.sort()
```

Comment:
Consider

```python
expected_types = sorted(literal_type for lt, literal_type in logical_types if lt == logical_type)
```

instead of sorting after the fact